pymongo/auth.pyi

```diff
@@ -0,0 +1,29 @@
+from typing import Any, Dict, FrozenSet, NamedTuple
+
+from pymongo.pool import SocketInfo
+
+MECHANISMS: FrozenSet[str] = ...
+
+class MongoCredential(NamedTuple):
+    mechanism: str
+    source: str
+    username: str
+    password: str
+    props: Any
+class GSSAPIProperties(NamedTuple):
+    service_name: str
+    canonicalize_host_name: bool
+    service_realm: Any
+def _build_credentials_tuple(mech: str, source: str, user: str, passwd: str, extra: Dict[str, Any]) -> MongoCredential: ...
```

Comment:
I don't think we should bother adding type hints to private functions/methods like this. Let's stick to things that are [publicly documented](https://github.com/mongodb/mongo-python-driver/tree/master/doc/api/pymongo).